qrand/platforms/qsharp/circuit.py

```diff
@@ -28,76 +28,134 @@
 ###############################################################################
 class QsharpCircuit(QuantumCircuit):
     def __init__(self, num_qubits: int) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+        self.gates: str = ""
```

Comment:
Make this into a property without a setter. That way people will be able to access `self.gates` but not edit it. Internally you will edit it by using the corresponding `self._gates` attribute.